### PR TITLE
Add settings for template file to open on creating a new file for Alias

### PR DESCRIPTION
**Pull request by staceyoue**
@barbara-darkshot @rob-aitchison 

I've defined the setting in the Alias engine, since this is only applies to Alias for now -- not sure if it is better to put it in the tk-multi-workfiles2 app instead?

The config setting value (e.g. data/templates/new_file/.../.wire) is relative to the tk-alias engine hooks/tk-multi-workfiles2 directory. I've added a `data` folder at the same level of the `basic` directory, to store the template Alias files -- let me know if you guys have any better suggestions on where to store the template files and/or how to store/retrieve the path to those files.

For now, Sebastien has only given template file for Modeling step - Exterior assets. We can add more as they come.

Here's the related tk-alias PR to go with this one:
https://github.com/shotgunsoftware/tk-alias/pull/68

### CODE REVIEW

File: env/includes/settings/tk-alias.yml [hunk 1]
```diff
@@ -70,4 +73,7 @@ settings.tk-alias.project:
     tk-multi-workfiles2: "@settings.tk-multi-workfiles2.alias"
   menu_favourites:
   - {app_instance: tk-multi-workfiles2, name: File Open...}
+  new_file_template:
```
Comment (barbara-darkshot) [inline env/includes/settings/tk-alias.yml:76]: I think you don't need to define the template for the project environment, as the "new_file" operation will only happen in the asset.step context (for me, the context is changed before calling the hook so this one should be executed in the right context but I may be wrong)

**Comment by barbara-darkshot** [comment]
hey @staceyoue and @rob-aitchison, 

for me, it's better to keep the changes to tk-alias and not move them to tk-multi-workfiles as this workflow is really Alias specific.

One thing I've thought, shouldn't we publish the alias reference file? Maybe into the Library project on a "Template" Asset? so we could follow it's evolution if we need to do some changes

I'm not sure putting this file into the config is the best idea because if the Alias lead wants to make some modification to the Alias file template, he will need an admin to push his changes to the config..

File: env/includes/settings/tk-alias.yml (hunk 1 above)
Comment (staceyoue) [reply env/includes/settings/tk-alias.yml:76]: We need to define the template for the project environment, an example where the project env is used is on opening Alias for the first time and on selecting a task in workfiles2 and then clicking "New File".

File: env/includes/settings/tk-alias.yml [hunk 2]
```diff
@@ -57,6 +57,9 @@ settings.tk-alias.asset_step:
   - {app_instance: tk-multi-snapshot, name: Snapshot...}
   - {app_instance: tk-multi-workfiles2, name: File Save...}
   - {app_instance: tk-multi-publish2, name: Publish...}
+  new_file_template:
+    Modeling:
+      Exterior: "data/templates/new_file/Modeling/Exterior/new_file_template.wire"
```
[the remark this answers is not in the document]
Comment (staceyoue) [reply env/includes/settings/tk-alias.yml]: I updated this to use a template string defined in templates.yml. The template string is a relative path from the tk-alias /hooks directory - to keep the template files stored in the alias engine vs. the config for everyone. Not sure this is the best way to store it and access, so let me know if you have other suggestions.

**Comment by rob-aitchison** [comment]
> I'm not sure putting this file into the config is the best idea because if the Alias lead wants to make some modification to the Alias file template, he will need an admin to push his changes to the config..

Hey @barbara-darkshot , I agree that it should end up as a Project-based PublishedFile (and/or workfile.)
Ideally, we can provide our starting versions of these files as examples that live in the config which can then get "transferred" to each customer site's Project folder somehow. Let's all discuss this together.
Thanks!

**Comment by barbara-darkshot** [comment]
hey @rob-aitchison , 
maybe we could include them in the schema like this example

image

**Comment by rob-aitchison** [comment]
@barbara-darkshot Excellent suggestion! 👍🏻 
Let's think about the name of the folder since `templates` is such an overloaded term within SG and SGTK.

**Comment by staceyoue** [comment]
So I've updated the config to store the new file templates for Alias in `schema/work_area/assets/template_files/alias`. I'm trying to define the name of the file as:
```
{sg_asset_type}_{Step}_new_file.wire
```
, but the template cannot resolve those fields. For example, as mentioned in the tk-alias PR I'm trying to get the fields using:
```
context.as_template_fields(template)
```
but it does not find any fields

**Pull request closed without merging**